src/components/metadata/metadata-row.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import MetaDataList from './metadata-list';
+import MetaDataValue from './metadata-value';
+import './styles/metadata.css';
+
+/**
+ * Shows metadata label and a given single value, or a list of values, or child elements
+ */
+const MetaDataRow = ({
+  label,
+  property,
+  kind = 'text',
+  empty = '-',
+  visible = true,
+  inline = true,
+  commas = true,
+  children
+}) => {
+  const showElements = React.isValidElement(children);
+  const showList = !showElements && Array.isArray(children) && children.length;
+  const showValue = !showList && !showElements;
+
+  return (
+    visible && (
+      <div className="pipeline-metadata__row">
+        <h3 className="pipeline-metadata__label">{label}</h3>
+        {showList && (
+          <MetaDataList
+            property={property}
+            inline={inline}
+            commas={commas}
+            kind={kind}
+            empty={empty}>
+            {children}
+          </MetaDataList>
+        )}
+        {showValue && (
+          <MetaDataValue value={children} kind={kind} empty={empty} />
+        )}
```

Comment:
```suggestion
        {showList ? (
          <MetaDataList
            property={property}
            inline={inline}
            commas={commas}
            kind={kind}
            empty={empty}>
            {children}
          </MetaDataList>
        ) : (
          <MetaDataValue value={children} kind={kind} empty={empty} />
        )}
```

Reply:
Again I wrote it this way as I felt it was clearer to read, since with both of these changes you now need to follow the logic through to understand all the potential cases here, rather then it being self describing. But I can still make the change if you think it's clear enough?

Reply:
Yeah I hear you - I just found that it was relatively straightforward to understand that if there's a list, show the list, else just show a value. The extra steps actually felt a bit less clear tbh as you need to puzzle out the extra expressions. I think it would make sense to include only if `showValue` were more complex than just `!showList`

Reply:
Actually trying this again I remember why I needed to do this originally, because there's a subtle requirement that this fails on, that a value or list should not be shown if there are React children.

I think there might be a better way to avoid this though by changing around some props instead, so I'll see if that works better.

Reply:
I've made that refactoring so now this should be avoided anyway.